omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java

```diff
@@ -54,11 +54,25 @@
                                      UiUtils ui,
                                      @SpringBean("emrApiProperties") EmrApiProperties emrApiProperties,
                                      @SpringBean("emrConceptService") EmrConceptService emrConceptService,
+                                     @SpringBean("conceptService") ConceptService conceptService,
                                      @RequestParam("term") String query,
+                                     @RequestParam(value = "diagnosisSetUuids", defaultValue = "") String diagnosisSetUuids,
                                      @RequestParam(value = "start", defaultValue = "0") Integer start,
                                      @RequestParam(value = "size", defaultValue = "50") Integer size) throws Exception {
 
-        Collection<Concept> diagnosisSets = emrApiProperties.getDiagnosisSets();
+        Collection<Concept> diagnosisSets = new ArrayList<Concept>();
+        if (StringUtils.isNotEmpty(diagnosisSetUuids) && !"undefined".equalsIgnoreCase(diagnosisSetUuids)) {
```

Comment:
The check on `undefined` shouldn't be here. It's rather a client-side effort to ensure that a valid string is sent here. If `undefined` lands here anyway then so be it.

Reply:
Yes, this is actually [catered for on RA-1603](https://github.com/openmrs/openmrs-module-coreapps/pull/199/files#diff-7d6c8132f77707ede46ad8ec6a618a59R78) but some js implementations may consider an **empty string** as `undefined`.

Reply:
So please remove that check, it not the place of a Java controller to take care of such JavaScript edge cases.

Reply:
Yes, I have cleared this.